### ISSUE TITLE
solarium sends float according to LC_ resulting in errors on SOlR

### ISSUE THREAD

**Issue by quazardous** [solariumphp/solarium#262]
when dealing with float values solarium send a converted string using LC_

so if I use LC_ = fr SolR gets the decimal separator , (comma) in the float value.

solarium should temporary force LC_ to "C" the set back the original value

**Comment by thomascorthals** [comment]
Meanwhile you can do it in your own code as a workaround. But it would be better for Solarium to take care of it because this is the kind of thing that can catch you unawares.

```php
<?php

// make sure floats use "." as decimal separator
$currentLocale = setlocale(LC_NUMERIC, 0);
setlocale(LC_NUMERIC, 'C');

/*
 * Send whatever requests you want Solarium to send.
 */

// restore the locale
setlocale(LC_NUMERIC, $currentLocale);

```

This workaround won't break things once Solarium takes care of it internally.

**Comment by quazardous** [comment]
@thomascorthals it's what I did :) OC thx anyway

**Comment by thomascorthals** [comment]
This issue is due to the way PHP handles float to string conversion internally. It's not something that can be "fixed" and coding around it inside Solarium would be a lot of work. On top of that, testing such an implementation isn't straightforward either. See discussion #909 for a full explanation.

The internal behaviour has been [fixed in PHP 8.0](https://wiki.php.net/rfc/locale_independent_float_to_string) and this workaround is no longer necessary from PHP 8 onward. Leaving it in place doesn't cause backward compatibility issues though.
